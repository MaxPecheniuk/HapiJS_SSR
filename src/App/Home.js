import React from 'react'


export default function Home() {
	return(
		<div>
			<h1>
				Lorem ipsum dolor  praesentium reprehenderit saepe.
			</h1>
		</div>
	)
}