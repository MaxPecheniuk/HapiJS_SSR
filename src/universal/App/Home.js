import React from 'react'
import './index.scss'

export default function Home() {
	return(
		<div className="red">
			<h1>
				!!!!!!Lorem ips555um dolor  praesentium reprehenderit saepe.
			</h1>
		</div>
	)
}