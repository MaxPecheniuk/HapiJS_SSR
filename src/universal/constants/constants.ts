export const CHANGE_LANGUAGE = 'CHANGE_LANGUAGE';
export const CHECK_LANGUAGE = 'CHECK_LANGUAGE';
// export const DELETE_TITLE = 'DELETE_TITLE';