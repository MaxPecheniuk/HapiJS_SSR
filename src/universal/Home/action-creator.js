import {COUNT} from "../constants/constants";

const count = (payload) => {
	return {type: COUNT, payload}
}
