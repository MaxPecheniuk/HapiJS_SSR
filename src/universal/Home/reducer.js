import {COUNT} from "../constants/constants";

const initialState = {
	count: 0
}

export const HomeReducer = (state = initialState, action) => {
	switch (action.type){
		case COUNT:
			return{}
		default: initialState
	}
}