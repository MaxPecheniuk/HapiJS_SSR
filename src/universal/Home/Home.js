import React from 'react';
import {connect} from 'react-redux';
import { decrease, increase } from './action-creator';
import '../index.scss';

const mapStateToProps = state => ({
  count: state.HomeReducer.count,
})
const mapDispatchToProps = {
  increase,
  decrease,
};

class Home extends React.Component{

  testClick = (a,b) => {
    return a +b;
  }

  render() {

		const { count, increase, decrease, title, url } = this.props;
    return (
      <div className="red">
        {count}
        <button className="test" onClick={increase}>+</button>
        <button onClick={decrease}>-</button>
        <a href={url} onClick={this.testClick}>{title}</a>
      </div>
    )
  }
}


export default connect(mapStateToProps, mapDispatchToProps)(Home)