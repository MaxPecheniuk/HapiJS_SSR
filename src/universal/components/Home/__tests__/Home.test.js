/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme/build';
import * as actions from '../action-creator';
import * as types from '../../../constants/constants'
import { Home } from '../Home';


describe('Home component', () => {
  let component;
  beforeEach(() => {
    component = shallow(<Home/>);
  })

  it('should render correctly with no props', () => {
    //only parent without child
    expect(component).toMatchSnapshot();
  });

  it('should be possible to activate button with spacebar and click', () => {
    component
      .find('.btn')
      .simulate('keydown', {keyCode: 32})
      .simulate('click');
    expect(component.state('showBlock')).toBe(true);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
  it('use jsdom in this test file', () => {
    const element = document.createElement('div');
    expect(element).not.toBeNull();
  });

})

describe('action creator', () => {
  it('should AC an action increase value', () => {
    const expectedAction = {
      type: types.INCREASE
    }
    expect(actions.increase()).toEqual(expectedAction)
  })
  it('should AC an action decrease value', () => {
    const expectedAction = {
      type: types.DECREASE
    }
    expect(actions.decrease()).toEqual(expectedAction)
  })
})
