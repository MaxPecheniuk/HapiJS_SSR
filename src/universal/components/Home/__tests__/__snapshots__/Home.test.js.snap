// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should be possible to activate button with spacebar and click 1`] = `ReactWrapper {}`;

exports[`Home component should render correctly with no props 1`] = `ShallowWrapper {}`;
